presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageDataSink.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.pagefile;
+
+import com.facebook.presto.spi.Page;
+
+import java.io.IOException;
+
+public interface PageDataSink
+{
+    /**
+     * Number of bytes written to this sink so far.
+     */
+    long size();
+
+    /**
+     * Gets the size of the memory buffers.
+     */
+    long getRetainedSizeInBytes();
+
+    /**
+     * Write a page
+     */
+    void write(Page dataPage)
```

Comment:
Also it feels like this method should accept `SerializedPage`. The `OrcDataSink` interface accepts `OrcDataOutput`, that is effective a "lazy" serialized representation. I don't think that we need the laziness in our case though.